### PR TITLE
Feature/compliance graph

### PR DESCRIPTION
**Pull request by mpgxvii**
Solves #285 and partially solves: #87

### CODE REVIEW

File: src/app/study/containers/study-page.component.html [hunk 1]
```diff
@@ -37,25 +37,41 @@
...
-          [isLoaded]="subjectsLoaded$ | async"
-          (openSubject)="openSubjectHandler($event)"
-        ></app-subject-table>
-        <app-subject-detail
-          *ngIf="subject"
-          [subject]="subject"
-        ></app-subject-detail>
+        <section
+          [class.subject-table]="subject"
+          [class.subject-table-before]="!subject"
```
Comment (nivemaham) [inline src/app/study/containers/study-page.component.html]: I find this [syntax](https://toddmotto.com/ng-class-angular-classes#multiple-classes-with-ngclass) easy to read.

File: src/app/study/components/compliance-plot/compliance-plot.component.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ import { ConfigKey } from '../../../shared/models/config.model'
   template: `
     <app-chart-base-multi-bar
       *ngIf="isLoaded && data && keys"
+      [margin]="{top: 40, right: 16, bottom: 32, left: 80}"
```
Comment (nivemaham) [inline src/app/study/components/compliance-plot/compliance-plot.component.ts:17]: Can't we move these to scss ?

File: src/app/study/components/subject-table/subject-table.component.ts [hunk 3]
```diff
@@ -25,12 +25,15 @@ export class SubjectTableComponent implements OnInit, OnDestroy {
...
     'humanReadableId',
-    'status',
+    'subjectStatus',
+    'deviceStatus',
     'sources',
     'lastSeen'
   ]
   dataSource: SubjectDataSource | null
   subjectDB = new SubjectDB()
+  statusCon = 'CONNECTED'
```
Comment (nivemaham) [inline src/app/study/components/subject-table/subject-table.component.ts]: what do we use these for? Aren't the status values read from response received?

File: src/app/study/components/compliance-plot/compliance-plot.component.ts (hunk 2 above)
Comment (mpgxvii) [reply src/app/study/components/compliance-plot/compliance-plot.component.ts:17]: This margin is the input for the d3 chart component itself. All charts use the same chart-base component so the same code is used to initialise the chart with different widths based on the margins inputted.

File: src/app/study/components/subject-table/subject-table.component.ts (hunk 3 above)
Comment (mpgxvii) [reply src/app/study/components/subject-table/subject-table.component.ts]: Yeah I initially used this for testing out the component. I'll remove this.

**Review by nivemaham — APPROVED** [review]
LGTM

**Pull request merged**